lib/galaxy/jobs/runners/drmaauniva.py

```diff
@@ -0,0 +1,640 @@
+    since we are only searching the last 2K
+    checks for an error message that indicates an memory constraint violation
+    returns True if such an indicator is found and False otherwise
+    """
+    # list of error output from different programming languages in case
+    # of memory allocation errors for bash, Python, C++, JAVA, Perl
+    memerrors = set(["xrealloc: cannot allocate",
+                     "MemoryError",
+                     "std::bad_alloc",
+                     "java.lang.OutOfMemoryError: Java heap space",
+                     "Out of memory!"])
+
+    try:
+        log.debug( 'Checking %s for exceeded memory messages from programs', efile_path )
+        with open( efile_path ) as f:
+            if os.path.getsize(efile_path) > MEMORY_LIMIT_SCAN_SIZE:
+                f.seek(-MEMORY_LIMIT_SCAN_SIZE, os.SEEK_END)
+                f.readline()
+            for line in f.readlines():
+                stripped_line = line.strip()
+                for err in memerrors:
+                    if err in stripped_line:
+                        return True
+    except:
+        log.exception('Error reading end of %s:', efile_path)
+
+    return False
+
+
+def _parse_mem( mstring ):
+    mem = None
+    m = re.search( "([0-9.]+)([KGM]?)", mstring )
+    if m is not None:
+        mem = float(m.group(1))
+        if m.group(2) == 'K':
+            mem *= 1024
+        elif m.group(2) == 'M':
+            mem *= 1024 * 1024
+        elif m.group(2) == 'G':
+            mem *= 1024 * 1024 * 1024
+        elif m.group(2) == '':
+            pass
+        else:
+            logging.error( "DRMAAUniva: unparsable memory spec {spec}".format(spec=mstring) )
+    return mem
+
+
+def _parse_time( tstring ):
+    tme = None
+    m = re.search( "([0-9:.]+)", tstring )
+    if m is not None:
+        timespl = m.group(1).split(':')
+        tme = float( timespl[-1] )  # sec
+        if len( timespl ) > 1:     # min
+            tme += float( timespl[-2] ) * 60
+        if len( timespl ) > 2:     # hour
+            tme += float( timespl[-3] ) * 3600
+        if len( timespl ) > 3:     # day
+            tme += float( timespl[-4] ) * 86400
+    return tme
+
+
+def _parse_native_specs( job_id, native_spec ):
+    """
+    determine requested run time and memory from native specs
+    native_spec (e.g. h_rt=01:00:02 -l h_vmem=1G) the native
+    job_id the job ID (only used for logging)
+    specification string passed to GE
+    return time,mem (or None,None if nothing found)
+    """
+    tme = None
+    mem = None
+    # parse time
+    m = re.search( "rt=([0-9:]+)[\s,]*", native_spec )
+    if m is not None:
+        tme = _parse_time( m.group(1) )
+        if tme is None:
+            logging.error( "DRMAAUniva: job {job_id} has unparsable time native spec {spec}".format(job_id=job_id, spec=native_spec) )
+    # parse memory
+    m = re.search( "mem=([0-9.]+[KGM]?)[\s,]*", native_spec )
```

Comment:
@bernt-matthias quick question, does Univa also have an `-xml` option format the output? This would make parsing way more stable than the regex searches. I know that a few SGE commands do support the `-xml` argument.